spec/sailfish_iptables_dbus_interface_description.xml

```diff
@@ -0,0 +1,490 @@
+<arg name="destination_service" type="s" direction="in"/>
+<arg name="protocol" type="u" direction="in"/>
+<arg name="operation" type="q" direction="in"/>
+<arg name="result" type="q" direction="out"/>
+<arg name="string" type="s" direction="out"/>
+</method>
+
+<!-- 
+	GetVersion:
+	@version: Version number of the iptables API
+	
+	Returns sailfish iptables API version
+-->
+<method name="GetVersion">
+<arg name="version" type="i" direction="out"/>
+</method>
+
+<!--
+	Initialize:
+	
+	Sent when plugin is loaded into Connection Manager to notify that iptables can be managed using this DBUS API.
+-->
+<signal name="Initialize"></signal>
+
+<!--
+	Shutdown:
+	
+	Sent when plugin is removed from Connection Manager.
+-->
+<signal name="Shutdown"></signal>
+
+<!--
+	IptablesTableCleared:
+	@table: The table that was cleared.
+	
+	Sent when iptables table was cleared from rules.
+-->
+<signal name="IptablesTableCleared">
+<arg name="table" type="s"/>
+</signal>
+
+<!--
+	IptablesChainsCleared:
+	@table: The table that was cleared from custom chains.
+	
+	Sent when iptables table was cleared from custom chains.
+-->
+<signal name="IptablesChainsCleared">
+<arg name="table" type="s"/>
+</signal>
+
+<!--
+	PolicyChanged:
+	@chain: Name of the chain that was changed.
+	@policy: New policy for the chain.
+	
+Sent when policy is changed for a chain
+-->
+<signal name="PolicyChanged">
+<arg name="chain" type="s"/>
+<arg name="policy" type="s"/>
+</signal>
+
+<!--
+	RuleChanged:
+	@table: Rule added to table
+	@chain: Rule added to chain in table
+	@target: Target for the rule (ACCEPT, DROP, REJECT, QUEUE, RETURN or user added custom chain name with sfos_ prefix
+	@source_ip: Rule source IP address
+	@destination_ip: Rule destination IP address
+	@source_port_start: Source port range start
+	@source_port_end: Source port range end
+	@destination_port_start: Destination port range start
+	@destination_port_end: Destination port range end
+	@protocol: Protocol as string
+	@operation: 0 = add, 1 = remove
+	
+Sent after a rule is succesfully added to or removed from iptables. Any of the string parameters can be empty "". If port is 0 then port is not added with the rule. If destination/source end port is 0 then port range is not utilized and the ports defined in destination/source start is only used.
+-->
+<signal name="RuleChanged">
+<arg name="table" type="s"/>
+<arg name="chain" type="s"/>
+<arg name="target" type="s"/>
+<arg name="source_ip" type="s"/>
+<arg name="destination_ip" type="s"/>
+<arg name="source_port_start" type="q"/>
+<arg name="source_port_end" type="q"/>
+<arg name="destination_port_start" type="q"/>
+<arg name="destination_port_end" type="q"/>
+<arg name="protocol" type="s"/>
+<arg name="operation" type="q"/>
+</signal>
+
+<!--
+	ChainChanged:
+	@table: Affected table name
+	@chain: Affected chain name
+	@operation: Type of change: ADD or REMOVE
+	
+Sent when chain was added to or removed from a table.
+-->
+<signal name="ChainChanged">
```

Comment:
Similar concern here. `ChainChanged` is not emitted when the chain is changed (i.e. rules are added or removed). But it is emitted when chains are being added or removed. I suggest separate `ChainAdded` and `ChainRemoved` signals.

Reply:
Chains can be added, removed and flushed. I'm not so keen on adding three different types of signals (one for each operation).